Source/Core/Core/Movie.cpp

```diff
@@ -216,11 +220,17 @@ void FrameUpdate()
   s_bPolled = false;
 }
 
+static void CheckMD5();
+static void GetMD5();
+
 // called when game is booting up, even if no movie is active,
 // but potentially after BeginRecordingInput or PlayInput has been called.
 // NOTE: EmuThread
-void Init()
+void Init(const BootParameters& boot)
 {
+  if (std::holds_alternative<BootParameters::Disc>(boot.parameters))
+    s_current_file_name = std::get<BootParameters::Disc>(boot.parameters).image_path;
```

Comment:
Unfortunately no, GetMD5 is called somewhere other than in the boot process (Init). BeginRecordingInput calls it :(

I will clear s_current_file_name and return from GetMD5/CheckMD5 if it's empty.